com.incquerylabs.v4md/src/main/com/incquerylabs/v4md/expressions/BinaryVQLExpression.java

```diff
@@ -51,7 +53,10 @@ public BinaryVQLExpression(ParameterizedExpressionDescriptor descriptor, Express
 	}
 
 	public Object getValue(Element sourceParameter, ValueContext context) throws Exception {
-		ViatraQueryEngine engine = ViatraQueryAdapter.getAdapter(project).get().getEngine();
+		ViatraQueryEngine engine = ViatraQueryAdapter.getAdapter(project).orElseThrow(()-> {
```

Comment:
I don't think it is a good idea to throw an exception in the getValue() method as it gets called by MagicDraw APIs and the exception trace would be highly visible in the logs. I think in such cases simply returning an empty match set (similar to when the Binary Query Result Provider couldn't be initialized) would be a better approach here.

@lunkpeter you have more experience with using these expressions, what do you think about its updates.

Reply:
I agree completely. The expression executor does not handle internal exception particularly well, resulting in internal error notifications all across the UI

Reply:
Previously the optional was able to throw an exception (it wasn't checked before the get was called) and I thought that a more specific exception would be more useful but I understand your point so now I handle the uninitialized engine instead of throwing exceptions.